utils/custom_errors.go

```diff
@@ -0,0 +1,37 @@
+package utils
+
+// ValidationError is an error that occurs during validation.
+type ValidationError struct {
+	msg string
+}
+
+func (e *ValidationError) Error() string { return e.msg }
+
+// NewValidationError returns a validation error.
+func NewValidationError(msg string) error {
+	return &ValidationError{msg}
+}
+
+// DatabaseAccessError is an error that occurs during database access.
+type DatabaseAccessError struct {
+	msg string
+}
+
+func (e *DatabaseAccessError) Error() string { return e.msg }
+
+// NewDatabaseAccessError returns a database access error.
+func NewDatabaseAccessError(msg string) error {
+	return &DatabaseAccessError{msg}
+}
+
+// StatusCodeForError returns an http status code for an error type.
+func StatusCodeForError(err interface{}) int {
```

Comment:
I think this function would make more sense in the web package, it's unlikely that it will be reused by other packages. Also, moving it out of here will prevent cyclical imports if you need to refer to error types that are not in the utils package.